core/rate_limiter.py

```
@@ -0,0 +1,251 @@
+                now,  # ARGV[1] - current time
+                self.rate_per_second,  # ARGV[2] - rate
+                self.burst_capacity,  # ARGV[3] - capacity
+                tokens  # ARGV[4] - requested tokens
+            )
+            return bool(result)
+        except Exception as e:
+            # On Redis errors, allow request to prevent blocking legitimate traffic
+            print(f"Rate limiter error: {e}")
+            return True
+
+    def get_retry_after(self, identifier: str = "global") -> int:
```

Ruff (0.14.10)

Unused method argument: `identifier`

(ARG002)